[DNM v2.5.9/v2.6] Add additional config options for fluentd

dbason opened the pull request:
This change should meet the requirements for https://github.com/rancher/rancher/issues/31807

The replicas setting may be one worth plumbing through to the UI, but I haven't done this yet.

For tolerations and nodeSelectors, fluentd will use the chart wide settings unless overridden with fluent specific settings.

nickgerace commented on packages/rancher-logging/generated-changes/overlay/templates/loggings/aks/logging.yaml:
```diff
@@ -31,6 +31,8 @@ spec:
     resources: {{- toYaml . | nindent 6 }}
   {{- end }}
   fluentd:
+    scaling:
+      replicas: {{ .Values.fluentd.replicas }}
```
This should be wrapped to ensured that replicas are populated

nickgerace commented on packages/rancher-logging/generated-changes/overlay/templates/loggings/eks/logging.yaml:
```diff
@@ -32,6 +32,8 @@ spec:
     resources: {{- toYaml . | nindent 6 }}
   {{- end }}
   fluentd:
+    scaling:
+      replicas: {{ .Values.fluentd.replicas }}
```
Same as above

nickgerace commented on packages/rancher-logging/generated-changes/overlay/templates/loggings/gke/logging.yaml:
```diff
@@ -31,6 +31,8 @@ spec:
     resources: {{- toYaml . | nindent 6 }}
   {{- end }}
   fluentd:
+    scaling:
+      replicas: {{ .Values.fluentd.replicas }}
```
Same as above

nickgerace commented on packages/rancher-logging/generated-changes/overlay/templates/loggings/k3s/logging-k3s-openrc.yaml:
```diff
@@ -38,6 +38,8 @@ spec:
       {{- toYaml . | nindent 6 }}
   {{- end }}
   fluentd:
+    scaling:
+      replicas: {{ .Values.fluentd.replicas }}
```
Same as above

nickgerace commented on packages/rancher-logging/generated-changes/overlay/templates/loggings/k3s/logging-k3s-systemd.yaml:
```diff
@@ -38,6 +38,8 @@ spec:
       {{- toYaml . | nindent 6 }}
   {{- end }}
   fluentd:
+    scaling:
+      replicas: {{ .Values.fluentd.replicas }}
```
Same as above

nickgerace commented on packages/rancher-logging/generated-changes/overlay/templates/loggings/rke2/logging-rke2-containers.yaml:
```diff
@@ -43,6 +43,8 @@ spec:
       {{- toYaml . | nindent 6 }}
   {{- end }}
   fluentd:
+    scaling:
+      replicas: {{ .Values.fluentd.replicas }}
```
Same as above

nickgerace commented on packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:
```diff
@@ -60,6 +60,8 @@ spec:
       {{- toYaml . | nindent 6 }}
   {{- end }}
   fluentd:
+    scaling:
+      replicas: {{ .Values.fluentd.replicas }}
```
Same as above

nickgerace commented:
Thanks for the PR btw @dbason :)
I have the same question as @paynejacob did, and made one comment that exists in the Logging CRs.
After those are addressed, this will more than likely be good to go!

dbason commented on packages/rancher-logging/generated-changes/patch/values.yaml.patch:
```diff
@@ -84,12 +84,15 @@
 +  tls:
 +    enabled: false
 +fluentd:
++  replicas: 1
```
If the scaling is null in the Fluentd spec the operator creates it and populates it with 1 replica:
https://github.com/banzaicloud/logging-operator/blob/e53255c84b2d4e1f86f6ac628cc080d716ca9fa9/pkg/sdk/api/v1beta1/logging_types.go#L233-L237

dbason commented:
@nickgerace @paynejacob I've updated the commit to address the feedback.  I've also added the bufferStorageVolume in response to a request on the original issue.

nickgerace reviewed (dismissed):
LGTM after squashing all commits and rebase. Like the `disablePvc` check too before allowing the `bufferStorageVolume` too

dbason commented:
I already rebased to fix the failing CI job, I'll squath the second commit in now.

nickgerace commented:
@dbason would you mind rebasing one more time?

nickgerace commented:
Relevant issues:
- https://github.com/rancher/rancher/issues/31807
- https://github.com/rancher/rancher/issues/28401

nickgerace commented on packages/rancher-logging/package.yaml:
```diff
@@ -1,6 +1,6 @@
 url: https://kubernetes-charts.banzaicloud.com/charts/logging-operator-3.9.4.tgz
 packageVersion: 00
-releaseCandidateVersion: 03
+releaseCandidateVersion: 04
```
This should actually be
```
packageVersion: 01
releaseCandidateVersion: 00
```
since this is a v2.6/v2.5.9 chart

dbason replied:
> @dbason would you mind rebasing one more time?

@nickgerace done, and the version updated :)

dbason commented:
> @dbason would you mind rebasing one more time?

Done, and the version is updated :)

nickgerace reviewed (approved):
LGTM. Let's wait for v2.5.8 release to merge

nickgerace commented:
I believe we are good to close this since https://github.com/rancher/charts/pull/1182 has been merged with @dbason as a co-committer
